data_construction/EMP_data_construction/format_metadata.R

```diff
@@ -0,0 +1,92 @@
+# script to format the extra metadata
+rm(list = ls())
+library(dplyr)
+source('paths.r')
+
+#load mapping file; this tells us which samples we want to look at
+map <- readRDS(emp_map_clean.path)
+
+#load the metadata files we need 
+study_632  <- read.csv(emp_study632_metadata.path)
+study_722  <- read.csv(emp_study722_metadata.path)
+study_808  <- read.csv(emp_study808_metadata.path)
+study_864  <- read.csv(emp_study864_metadata.path)
+study_895  <- read.csv(emp_study895_metadata.path)
+study_990  <- read.csv(emp_study990_metadata.path)
+study_1031 <- read.csv(emp_study1031_metadata.path)
+study_1037 <- read.csv(emp_study1037_metadata.path)
+study_1043 <- read.csv(emp_study1043_metadata.path)
+study_1521 <- read.csv(emp_study1521_metadata.path)
+study_1579 <- read.csv(emp_study632_metadata.path)
+study_1674 <- read.csv(emp_study1674_metadata.path)
+study_1702 <- read.csv(emp_study1702_metadata.path)
+study_1716 <- read.csv(emp_study1716_metadata.path)
+study_1747 <- read.csv(emp_study1747_metadata.path)
+
+#make a list of the metadata files
+data.list <- list(study_632,  study_722,  study_808,  study_864,  study_895,  study_990,  
+                  study_1031, study_1037, study_1043, study_1521, study_1579, study_1674, 
+                  study_1702, study_1716, study_1747)
+
+#isolate the samples we want to look at
+samples <- map$`#SampleID`
+
+for(i in 1:length(data.list)){
+  data.list[[i]] <- subset(data.list[[i]], data.list[[i]]$sample_name %in% samples)
+}
+
+#make column names consistent
+for(i in 1:length(data.list)){
+    colnames(data.list[[i]])[colnames(data.list[[i]]) == 'carb_nitro_ratio'] <- 'c_n_ratio'
+    colnames(data.list[[i]])[colnames(data.list[[i]]) == 'perc_nitrogen'] <- 'percent_n'
+    colnames(data.list[[i]])[colnames(data.list[[i]]) == 'perc_total_c'] <- 'percent_c'
+    if('tot_carb' %in% colnames(data.list[[i]]) & 'tot_org_carb' %in% colnames(data.list[[i]])){
+      data.list[[i]] <- data.list[[i]][-data.list[[i]]$tot_org_carb]
+    }
+    if('tot_carb' %in% colnames(data.list[[i]]) & 'percent_c' %in% colnames(data.list[[i]])){
+      data.list[[i]] <- data.list[[i]][-data.list[[i]]$tot_carb]
+    }
+}
+
+#make units consistent
+data.list[[1]]$tot_nitro     <- data.list[[1]]$tot_nitro / 1000 * 100 # g N / kg -> kg N / kg soil -> percent N
+data.list[[1]]$tot_org_carb  <- data.list[[1]]$tot_org_carb / 1000 * 100 # g C / kg -> kg C / kg soil -> percent C
+colnames(data.list[[1]])[colnames(data.list[[1]]) == "tot_nitro"]      <- "percent_n"
+colnames(data.list[[1]])[colnames(data.list[[1]]) == "tot_org_carb"]   <- "percent_org_c"
+#3.   808: water content: ???
+data.list[[3]]$tot_carb      <- data.list[[3]]$tot_carb / 1000 * 100 
+data.list[[3]]$tot_nitro     <- as.numeric(data.list[[3]]$tot_nitro) / 1000 * 100 ## Warning: / not meaningful for factors
```

Comment:
Sometimes when you load data from other sources, formatting stuff happens, and numeric columns get interpreted as "factors". A good solution is to use a for-loop or a lapply call to transform any column that is supposed to be numeric to numeric. If its already numeric, nothing will change, but if it was read in as a character or factor vector then it will be transformed into a numeric column.